platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/iss/DefaultIssHandler.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2022-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.state.iss;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.common.merkle.utility.SerializableLong;
+import com.swirlds.common.scratchpad.Scratchpad;
+import com.swirlds.platform.components.common.output.FatalErrorConsumer;
+import com.swirlds.platform.config.StateConfig;
+import com.swirlds.platform.system.SystemExitCode;
+import com.swirlds.platform.system.state.notifications.IssNotification;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+/**
+ * A standard implementation of {@link IssHandler}.
+ */
+public class DefaultIssHandler implements IssHandler {
```

Comment:
If we move this class directly in a `com.swirlds.platform.state.iss.internal` package we can split better between public and private API

Reply:
Moved, although I should point out that the current refactoring effort is not attempted to address the package structure for platform api/impl. Right now package structure is very ad hoc, and I don't think we have a clear plan for how we want to structure things yet.